### ISSUE TITLE
Set up a CI/CD platform

### ISSUE THREAD

**Issue by BrooksPatton** [brooks-builds/bbggez#3]
Set up a CI/CD platform for testing and deploying the library when merged into master

**Comment by baileyn** [comment]
I have set up a simple CI using GitHub Actions to do the following:
* Check the project for errors
* Run all of the tests
* Check that the project has been formatted with `rustfmt`.

You can see it in action [here](https://github.com/baileyn/bbggez/actions). It currently takes around 7 minutes to run the check/test (they run in parallel). This is mainly due to having to grab the dependencies for the OS then having to compile the entire project. 

If you'd rather use another platform for the CI/CD, just let me know and I can work on getting it set up.

**Comment by baileyn** [comment]
@BrooksPatton I wasn't sure the best way to reach out to you. I saw on the stream that you weren't sure how to use the CI. If you click the title, in this case `Rust` (next to the green check mark or red x), you can see more information about the deployment. This allows you to see the steps that were taken. 

On the left, it'll show you all of the jobs that were in the workflow. In this case, there's `Check for errors`, `Test project`, and `Check formatting`. 

On the right, you can see all of the steps that are taken to complete the job you have selected. If you click on one of the steps, it'll expand, and you can see the logs for the step execution. Additionally, on the right side where the step is listed it shows how long that step took in seconds.

Lastly, the reason you were unable to see GitHub Actions on the top of your repository is because it's currently in beta. I believe you have to opt-in to GitHub Actions located [here](https://github.com/features/actions). Also, the reason you didn't see my Github Workflow file was because it was in the `github_actions` branch. You can view the file [here](https://github.com/baileyn/bbggez/blob/github_actions/.github/workflows/rust.yml). I don't like to commit any work to the master branch when I'm developing until it's merged in upstream.

**Comment by BrooksPatton** [comment]
I just got approved to start setting up actions! I'll take a look through it as soon as I get some good coding time.
